### PR TITLE
feat(ui-react): add admin user management pages

### PR DESCRIPTION
**Pull request by luizhf42**
## What

Admin user list and detail pages for the React UI, covering CRUD operations, search, pagination, login-as-user impersonation, and SAML password reset.

## Why

Implements features 3 and 4 from the admin panel roadmap (shellhub-io/team#93): User Administration and User Password Reset. The admin sidebar already links to `/admin/users` but the pages didn't exist yet.

## Changes

- **hooks/useAdminUsers**: Paginated list query with base64-encoded username search filter, single-user detail query. Both are admin-gated and skip 401 retries.
- **hooks/useAdminUserMutations**: Create, update, delete, and reset-password mutations. All invalidate relevant TanStack Query cache keys on success — reset-password invalidates both list and detail caches so the "Set Password" button disappears after enabling local auth on a SAML-only user.
- **api/errors**: Added `isSdkErrorWithBody` type guard to replace unsafe double-casts (`err as unknown as { body: string[] }`) when handling 400/409 responses with field-level error arrays.
- **pages/admin/users/index**: User list with debounced server-side search, row-click navigation to detail, inline edit/delete/login-as actions. Login-as opens an impersonation token URL in a new tab.
- **pages/admin/users/UserDetails**: Two-column info grid (identity + account), breadcrumb nav, edit drawer, delete confirmation, and a "Set Password" button (visible only for SAML-only users). Uses `type UserDetail = GetUserResponse & { admin?: boolean }` to work around the `admin` field being absent from the OpenAPI spec for the detail endpoint.
- **pages/admin/users/CreateUserDrawer, EditUserDrawer**: Form drawers with namespace limit controls (enable → disable creation or set max), password visibility toggle, confirmed/admin checkboxes. Edit drawer prevents self-demotion and un-confirming already-confirmed users.
- **pages/admin/users/ResetPasswordDialog**: Two-step dialog — confirm intent, then display generated password with copy button. Differentiates 400 ("already has local password") from generic errors.
- **App.tsx**: Registered `/admin/users` and `/admin/users/:id` routes inside `AdminRoute > LicenseGuard`.

## Testing

1016 tests passing (162 new). Coverage includes hook behavior (query guards, cache invalidation, filter encoding), type guard edge cases, and component interaction flows (form submission, error states, state reset on reopen, accessibility attributes).

### CODE REVIEW

File: ui-react/apps/console/src/pages/admin/users/index.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+      />
+
+      {/* Delete Confirmation */}
+      <ConfirmDialog
+        open={!!deleteTarget}
+        onClose={() => setDeleteTarget(null)}
+        onConfirm={async () => {
+          await deleteUser.mutateAsync({ path: { id: deleteTarget!.id } });
+          setDeleteTarget(null);
+        }}
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/pages/admin/users/index.tsx]: Addressed in d2cc5dc23. Added error handling to both delete callbacks (try/catch with inline error message) and login-as flows (button turns red with retry prompt on failure). Also replaced the `deleteTarget!` non-null assertion with an early return guard.

File: ui-react/apps/console/src/pages/admin/users/index.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+                      <code className="text-2xs font-mono text-text-muted">
+                        {user.username}
+                      </code>
+                    </td>
+                    <td className="px-4 py-3.5">
+                      <UserStatusChip
+                        confirmed={
+                          (user as unknown as { status?: string }).status
+                          === "confirmed" || user.confirmed === true
+                        }
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/pages/admin/users/index.tsx]: Won't address. The OpenAPI spec is outdated. The API doesn't provide the `confirmed` status anymore, but a `status: string` field, as described in #6088. The typing workaround will be removed as soon as the spec is fixed.

File: ui-react/apps/console/src/pages/admin/users/index.tsx [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+                          {loginAsId === user.id ? (
+                            <span className="w-4 h-4 border-2 border-primary/30 border-t-primary rounded-full animate-spin block" />
+                          ) : (
+                            <ArrowRightStartOnRectangleIcon className="w-4 h-4" />
+                          )}
+                        </button>
+                        <button
+                          onClick={(e) => {
+                            e.stopPropagation();
+                            setDeleteTarget(user);
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/pages/admin/users/index.tsx:222]: Addressed in cc99ca3d1. Clearing `deleteError` when opening the dialog in both list and detail pages.
